experiment/revert-bump.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -o errexit
+set -o nounset
+set -o pipefail
+
+bold_blue() {
+  echo -e "\x1B[1;34m${*}\x1B[0m"
+}
+
+bold_yellow() {
+  echo -e "\x1B[1;33m${*}\x1B[0m"
+}
+
+main() {
+  bold_blue "Checking out latest upstream ..."
+  git fetch https://github.com/kubernetes/test-infra master
+  git checkout FETCH_HEAD
+
+  bold_blue "Finding most recent bump commit ..."
+  local most_recent_bump
+  local most_recent_bump_commmit
+  most_recent_bump="$(git log --no-merges --extended-regexp \
+    --grep='Update prow to v[a-f0-9]+-[a-f0-9]+, and other images as necessary\.' \
+    --pretty=oneline -n 1)"
+  bold_yellow "Matching Commit: ${most_recent_bump}"
+  most_recent_bump_commmit=$(cut -f1 -d ' ' <<< "${most_recent_bump}")
+  echo "Commit SHA: ${most_recent_bump_commmit}"
+
+  bold_blue "Creating revert ..."
+  local revert_branch
+  revert_branch="revert-$(date +'%s')-${most_recent_bump_commmit}"
+  git checkout -b "${revert_branch}"
+  git revert --no-edit "${most_recent_bump_commmit}"
+
+  bold_blue "Pushing branch ..."
+  git push
```

Comment:
Best to be explicit here since if the user has `push.default` set to, for example, `simple` (*default*)  or `upstream` this will exit earlt with something like: 
```
fatal: The current branch revert-1569475888-5b1c4a484d7f828135be09a9aad9f365baf60488 has no upstream branch.
```

Instead:
```suggestion
  git -c push.default=current push
```

Reply:
good catch, done!

Reply:
WAIT, I CAN CONFIGURE THIS???

Reply:
Apparently, https://git-scm.com/docs/git-config#Documentation/git-config.txt-pushdefault

Reply:
yeah I have that by default in my dotfiles and totally forgot about this :-)
you never have to think about this again ... except when you script git

Reply:
```console
$ git remote -v
origin	https://github.com/fejta/repo-infra.git (fetch)
origin	git@github.com:fejta/repo-infra.git (push)
upstream	https://github.com/kubernetes/repo-infra.git (fetch)
upstream	no-push (push)
```
How does this ensure I push to origin rather than upstream?

Reply:
`origin` is the default if unspecified.

If you setup `origin` to be the upstream and call your fork something else then that doesn't help but in that case we can't do much besides something over-engineered parsing your remotes.

We're also ensuring the branch is some auto-generated name rather than say `master` so if it did get pushed it's not a huge deal.

Ideally nobody should have a push remote pointed upstream (IE have your safeguard), if we ever actually needed to push to the upstream we can do `git push https://github.com/kubernetes/test-infra.git ...`